File: Spring 2020/TwoSum/Python/main.py
def two_sum(a,b):
    look_up_table = {}

    for i in a:
        if i in look_up_table:
            return True
        look_up_table[b - i] = i

    return False

def two_sum_brute_force(a, b):
    for i in range(len(a)):
        for j in range(i + 1, len(a)):
            if a[i] + a[j] == b:
                return True

    return False

File: Spring 2020/TwoSum/Java/main.java
import java.util.Hashtable;

public class main {
    public static void main(String []args) {
        int[] x = {-2, 1, 2, 3, 7, 11};
        int y = 6;
        System.out.println(two_sum(x, y));
        System.out.println(two_sum_brute_force(x, y));
    }
    
   public static boolean two_sum(int []input, int target){
      Hashtable<Integer, Integer> lookUpTable = new Hashtable<Integer, Integer>(); 
      for (Integer i : input) {
          lookUpTable.put(target - i, i);
      }
      
      for (Integer i : input) {
          if (lookUpTable.containsKey(i)) {
              return true;
          }
      }

      return false;
   }

   public static boolean two_sum_brute_force(int []input, int target) {
       for (int i = 0; i < input.length; i++) {
           for (int j = 0; j < input.length; j++) {
               if (input[i] + input[j] == b) {
                   return true;
               }
           }
       }

       return false;
   }
 }
